.github/workflows/test-lang-rust-ci.yml

```diff
@@ -59,19 +60,27 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Cache Cargo
+        id: cache-cargo
         uses: actions/cache@v3
         with:
           # these represent dependencies downloaded by cargo
           # and thus do not depend on the OS, arch nor rust version.
           path: ~/.cargo
-          key: cargo-cache1-
+          key: ${{ runner.os }}-cargo-cache1-${{ matrix.target }}-${{ matrix.rust }}-${{ hashFiles('**/Cargo.lock') }}
+
       - name: Cache Rust dependencies
+        id: cache-target
         uses: actions/cache@v3
         with:
           # these represent compiled steps of both dependencies and avro
           # and thus are specific for a particular OS, arch and rust version.
-          path: ~/target
-          key: ${{ runner.os }}-target-cache1-${{ matrix.rust }}-
+          path: lang/rust/target
+          key: ${{ runner.os }}-target-cache1-${{ matrix.target }}-${{ matrix.rust }}-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Cache hit ?!
```

Comment:
Do we really need this job?
We can check cache hit another way.
https://github.com/apache/avro/actions/runs/6393032720/job/17352024653#step:3:22

Reply:
No, this is just to see that it hits the cache. It will be removed before merging